pytensor/tensor/nlinalg.py

```diff
@@ -1048,6 +1049,8 @@ def kron(a, b):
     They don't have the same shape and order when
     a.ndim != b.ndim != 2.
 
+    This new function now works for ndim > 2
```

Comment:
Delete this whole note, we have a correct implementation now and it's just confusing.